### ISSUE TITLE
Multiple unit cells or lattice parameters example

### ISSUE THREAD

**Issue by vamatya** [FABLE-3DXRD/xrd_simulator#15]
Hi, 

For powdered sample (2D) XRD, is there an example with multiple unit cells/lattice parameters as input?

Thanks!

**Comment by AxelHenningsson** [comment]
Hi!

It would be helpful if you could specify more details about what you want.

2D - in what sense? 
Multiple unit cells - like a two phase material?

What part of your question is not covered by the documentation?

Cheers 
Axel Henningsson

**Comment by vamatya** [comment]
Hi Axel, 

Thanks for the prompt reply. 

Yes, I'm looking for an example where I can define a material with multiple phases.  

RE 2D: I meant examples generating 2D diffraction rings.

I'm using `polycrystal_from_odf(..)` function for the above, as shown in the "s3dxrd_odf_sample.py". As per the documentation, I believe multiple phases can be defined for `Polycrystal(...)`- is this the only way to define two(/multiple) phase material?

Thanks,
Vinay

**Comment by AxelHenningsson** [comment]
Thank you. :=)

This amounts to a simple generalisation of `polycrystal_from_odf()` [at these lines](https://github.com/FABLE-3DXRD/xrd_simulator/blob/21e58b6b3f83bb174046416791e057b92b459e68/xrd_simulator/templates.py#L172)

the following code

````python
    # Sample is uniformly single phase
    phases = [Phase(unit_cell, sgname, path_to_cif_file)]
    element_phase_map = np.zeros((mesh.number_of_elements,)).astype(int)
````

would change as

````python
    # Sample is multi-phase with each phase equally present (volume fractions the same)
    phases  = [ Phase(uc, sg, path) for uc, sg, path in zip(unit_cell, sgname, path_to_cif_file)]
    element_phase_map  = np.random.randint(len(phases  ), size=(mesh.number_of_elements,))    
````

The input arguments of  `polycrystal_from_odf()` (`unit_cell`, `sgname` and `path_to_cif_file`) would then need to become iterables.

Perhaps this is something you would like to contribute with? Feel free to open a pull-request.

Cheers
Axel

**Comment by vamatya** [comment]
Thanks Axel! :)

I'll try that, and create a pull-request.

**Comment by AxelHenningsson** [comment]
Cool. Let me know if you run into any thing strange - and I will try to support. 👍 

Cheers
Axel

**Comment by vamatya** [comment]
Axel, 

For the 2 phase in the above example, how can one assign phase fraction, eg. 30%,70%?

Thanks,
Vinay

**Comment by vamatya** [comment]
I think I'll need to change this: 
`element_phase_map  = np.random.randint(len(phases  ), size=(mesh.number_of_elements,))`

**Comment by AxelHenningsson** [comment]
Now merged in #16 😎 

Great work ! ❤️ 

Feels great to have people engaged to make this software better and better. 🚀 

Cheers
Axel
